gistUpload.js

```diff
@@ -13,23 +13,30 @@ const HANG_PERCENT = 0.25;
  * @param {string} content - Contents of file from readFile
  * @param {string} gist - Name of gist supplied by user
  */
-function uploadFile(content, gist) {
-  return axios({
-    method: 'post',
-    url: API_URL,
-    auth: {
-      user: process.env.USER,
-      password: process.env.SECRET_TOKEN,
-    },
-    data: { files: { [gist]: { content } } },
-  })
-    .then((response) => {
-      console.log(`\nYour gist is done uploading.\nView it at: ${response.data.html_url}`);
-      return (response.data.html_url);
-    })
-    .catch(error => (console.log(error.response.data)));
+function uploadFile(contents, gist) {
```

Comment:
This should be an async function, and validations should be outside of the `try / catch`. smth like:

```javascript

async function ... {
  if (!gist || ...) {
    throw new Error(...);
  } else if ( ... ) {
    ...
  }

  try {
    const response = await axios.post( ... );

    console.log(`\nYour gist is done uploading.\nView it at: ${response.data.html_url}`);
    return (response.data.html_url);

  } catch (error) {
    // if you want to catch the error here, you'll need the try/catch block ...
  }
}

Reply:
done